File: sources/v1/server/src/main/java/com/furongsoft/agv/schedulers/entities/Material.java
package com.furongsoft.agv.schedulers.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

/**
 * 物料
 *
 * @author Alex
 */
@Getter
@Setter
@AllArgsConstructor
public class Material {
    /**
     * 物料编码
     */
    private String code;

    /**
     * 数量
     */
    private Integer count;
}

File: sources/v1/FrogAgvScanner2/app/src/main/java/com/furongsoft/frogagvscanner/entity/Material.kt
package com.furongsoft.frogagvscanner.entity

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.IgnoredOnParcel
import kotlinx.android.parcel.Parcelize

/**
 * 原料
 *
 * @author alex
 */
@Parcelize
class Material(
    @SerializedName("code") var code: String,
    @SerializedName("name") var name: String,
    @SerializedName("count") var count: Int
) : Parcelable {
    interface OnMaterialClickListener {
        fun onClick(material: Material)
    }

    @IgnoredOnParcel
    var listener: OnMaterialClickListener? = null
}
